src/components/Overview/Parsing.tsx

```diff
@@ -0,0 +1,514 @@
+const FilesContent = [
+  PrimaryLocation,
+  PrimaryPublicLocation,
+  GroupInteractions,
+  OffFacebookActivity,
+  AdvertisersUsingYourInfos,
+  AdsInterests,
+  FriendsPeerGroup,
+  UserPosts,
+  RecentlyViewed,
+  HangoutsConversations,
+  HomeAppData,
+  HomeAppHistory,
+  PeopleInteractions,
+  AdvertisersYouInteractedWith,
+  CommentsPosted,
+  RecentlyVisited,
+  EventResponses,
+  AccountsCenter,
+  PostsGroups,
+  CoverPhotosPosted,
+  ProfileInformation,
+  EventsInteractions,
+  AppsAndWebsites,
+  YourApps,
+  PostsAndComments,
+  EventsInvitations,
+  InstantGames,
+  PaymentHistory,
+  FriendRequestsReceived,
+  FriendRequestsSent,
+  FriendRequestsRejected,
+  WhoYouFollow,
+  CreatorBadges,
+  CommentsInGroups,
+  GroupMembershipActivity,
+  YourGroups,
+  Timezone,
+  AddressBooks,
+  FollowedPages,
+  LikedPages,
+  UnfollowedPages,
+  RecommendedPages,
+  YourPages,
+  ProfilePhotosPosted,
+  JournalPhotosPosted,
+  UserVideos,
+  PrivacyCheckupInteractions,
+  SavedItems,
+  SearchHistory,
+  AccountActivity,
+  AuthorizedLogins,
+  IpAddressActivity,
+  LoginProtectionData,
+  LoginsLogouts,
+  MobileDevices,
+  WhereYouLoggedIn,
+];
+
+// Create a list with the names of properties that will be used to describe each data point retrieved in the files uploaded by the user, based on the DataModel.
+const ObjectPropertiesName = [
+  'action_type',
+  'data_origin',
+  'data_type',
+  'platform',
+];
+
+// Create a random plot to use console.log()
+const BarChart = () => {
+  return (
+    <Plot
+      data={[
+        {
+          type: 'bar',
+          x: [
+            'Location',
+            'Behavioural',
+            'Communication',
+            'Technical',
+            'Social relationship',
+            'Contact',
+            'Transactional',
+          ],
+          y: [1, 2, 3, 3, 2, 3, 1],
+          marker: {
+            color: [
+              '#636EFA',
+              '#EF553B',
+              '#00CC96',
+              '#AB63FA',
+              '#FFA15A',
+              '#19D3F3',
+              '#FF6692',
+            ],
+          },
+        },
+      ]}
+      layout={{
+        width: 800,
+        height: 450,
+        title: 'What is being tracked ?',
+        paper_bgcolor: 'rgba(0,0,0,0)',
+        plot_bgcolor: 'rgba(0,0,0,0)',
+        yaxis: { title: { text: 'Number of data points', font: { size: 13 } } },
+        xaxis: { categoryorder: 'total descending' },
+      }}
+    />
+  );
+};
+
+// Initiation of the main function that will be used to parse the content of the files uploaded by the user.
+let SmartParser = (
+  FilesUploaded: any,
+  FilesContent: any,
+  DataModel: any,
+  ObjectPropertiesName: any
+) => {
+  // Pre-select the files for test purposes (step by step tests to debug specific files)
+  const FileUploaded = FilesUploaded.slice(0, 56); // 56 is the max length (all files have been tested)
```

Comment:
A supprimer car tu prendras en input qu'un seul fichier

Reply:
Oui je garde juste pour le moment en attendant de connecter avec ta partie (ça me permet de faire mes tests). Par contre, j'ai bien retiré l'itération sur la liste de fichiers dans la fonction.